exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -202,6 +223,19 @@ func (r *AzureManagedMachinePool) validateMaxPods() error {
 	return nil
 }
 
+func (r *AzureManagedMachinePool) validateOSDiskType() error {
+	if r.Spec.OsDiskType != "" {
+		if !(r.Spec.OsDiskType == string(containerservice.OSDiskTypeEphemeral) || r.Spec.OsDiskType == string(containerservice.OSDiskTypeManaged)) {
```

Comment:
that can be delegated to openAPI schema validation

Reply:
@CecileRobertMichon do you agree this webhook validation is superfluous if we add the enum allowed values to the api property spec?

Reply:
+1, for simple enumeration validation, it's better to use openAPI schema validation and only use webhook validation for more complex validation.

Reply:
done